pkg/processor/template_utils.go

```diff
@@ -0,0 +1,67 @@
+// Copyright © 2024 Meroxa, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package processor
+
+import (
+	"bytes"
+	"fmt"
+	"strconv"
+	"text/template"
+
+	"github.com/Masterminds/sprig/v3"
+	"github.com/conduitio/conduit/pkg/record"
+)
+
+// TemplateUtils provides methods to parse, execute go templates for provided records, and return the boolean value of the output.
+type TemplateUtils struct {
+	condition string
+	rec       record.Record
+	tmpl      *template.Template
+}
+
+func NewTemplateUtils(condition string, rec record.Record) *TemplateUtils {
+	return &TemplateUtils{
+		condition: condition,
+		rec:       rec,
+	}
+}
+
+// parse parses the provided template with injecting `sprig` functions, returns an error if parsing failed.
+func (t *TemplateUtils) parse() error {
+	tmpl, err := template.New("").Funcs(sprig.FuncMap()).Parse(t.condition)
+	if err != nil {
+		return err
+	}
+	t.tmpl = tmpl
+	return nil
+}
+
+// execute executes the template for the provided record, and parses the template output into a boolean, returns an error
+// if output is not a boolean, or if template is nil (should be called after TemplateUtils.parse).
```

Comment:
> should be called after TemplateUtils.parse

Why don't we prevent wrong use of the type and just parse the template when creating `TemplateUtils`? In other words, if `NewTeamplateUtils` returns no error you are sure you can use the returned type to evaluate conditions.